### PR TITLE
Fix result limit and sort templates.

### PR DESCRIPTION
**Pull request by EreMaijala**
This has been spun off from #2929. Makes the limit template similar to sort template.

### CODE REVIEW

**Comment by EreMaijala** [comment]
Mink tests are passing for me.

**Review by demiankatz — CHANGES_REQUESTED** [review]
I agree that changing the limit shouldn't reset the sort -- that seems like a possible copy and paste bug. But looking into this raised another issue: should changing both sort and limit reset the page number? Right now, if you use these controls on page 2, you remain on page 2, even though the meaning of "page 2" has been totally redefined by the changed parameter.

Maybe that's an issue better solved separately -- I think the fixes here make things better even without fixing that problem -- but I thought I'd leave this open until we could discuss. Let me know if you think it's worth tackling that here or starting a separate PR.

**Comment by demiankatz** [comment]
I should also note that this PR is going to conflict with #2814 when we get that far, but I agree that solving this for the current dev branch in the meantime still makes sense. We should just make sure that any changed behavior/fixes get correctly merged into that other PR after this work is completed.

**Comment by EreMaijala** [comment]
@demiankatz I decided to add filtering of the page parameter in both templates, now here to avoid having to revisit this later. :)

**Review by demiankatz — CHANGES_REQUESTED** [review]
As discussed in Slack, this looks good to me, but I'll hold off on merging until you've had time to add tests.

**Comment by EreMaijala** [comment]
Tests now added.

File: module/VuFind/tests/integration-tests/src/VuFindTest/Mink/SearchLimitTest.php [hunk 1]
```diff
@@ -67,7 +69,7 @@ protected function setUpLimitedSearch(
         $config = ['default_limit' => $default, 'limit_options' => $options];
         $this->changeConfigs(['searches' => ['General' => $config]]);
         $session = $this->getMinkSession();
-        $session->visit($this->getVuFindUrl() . '/Search/Results?limit=' . $limitParam);
+        $session->visit($this->getVuFindUrl() . '/Search/Results?lookfor=%22test publication%22&limit=' . $limitParam);
```
Comment (demiankatz) [inline module/VuFind/tests/integration-tests/src/VuFindTest/Mink/SearchLimitTest.php]: Should we perhaps add the appropriate building filter here to limit to a single MARC file, just to ensure that future records are less likely to break the test?

File: module/VuFind/tests/integration-tests/src/VuFindTest/Mink/SearchSortTest.php [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+                    ],
+                    'Sorting' => [
+                        'title' => 'Title',
+                        'title desc' => 'Title Reversed',
+                    ],
+                ],
+            ]
+        );
+        $session = $this->getMinkSession();
+        $session->visit($this->getVuFindUrl() . "/Search/Results?lookfor=%22test publication%22&sort=$sortParam");
```
Comment (demiankatz) [inline module/VuFind/tests/integration-tests/src/VuFindTest/Mink/SearchSortTest.php]: Same here...

**Review by demiankatz — CHANGES_REQUESTED** [review]
Thanks, @EreMaijala -- one small suggestion to hopefully make this more future-proof. I was hoping to make the change myself and test it today, but I never quite got there before running out of time!

File: module/VuFind/tests/integration-tests/src/VuFindTest/Mink/SearchLimitTest.php (hunk 1 above)
Comment (EreMaijala) [reply module/VuFind/tests/integration-tests/src/VuFindTest/Mink/SearchLimitTest.php]: Makes sense. I replaced the search phrase with a building filter.

File: module/VuFind/tests/integration-tests/src/VuFindTest/Mink/SearchSortTest.php (hunk 2 above)
Comment (EreMaijala) [reply module/VuFind/tests/integration-tests/src/VuFindTest/Mink/SearchSortTest.php]: Replaced here as well.

**Review by demiankatz — APPROVED** [review]
Thanks, @EreMaijala!

**Pull request merged**